runtime/src/snapshot_utils.rs

```diff
@@ -2814,4 +2820,211 @@ mod tests {
 
         assert_eq!(*bank4, roundtrip_bank);
     }
+
+    /// Test that cleaning works well in the edge cases of zero-lamport accounts and snapshots.
+    /// Here's the scenario:
+    ///
+    /// slot 1:
+    ///     - send some lamports to Account1 (from Account2) to bring it to life
+    ///     - take a full snapshot
+    /// slot 2:
+    ///     - make Account1 have zero lamports (send back to Account2)
+    ///     - take an incremental snapshot
+    ///     - ensure deserializing from this snapshot is equal to this bank
```

Comment:
I think a separate good test case is to have some additional, non-zero lamport updates to `Account1`, and ensure that the non-zero lamport update is cleaned up by clean, even though the zero lamport account itself wasn't cleaned

Reply:
Sure thing! Can you point me to code/docs for checking if updates are cleaned up?

Also, maybe it makes sense to put this as a test in AccountsDb, because these checks don't sound related to snapshots to me, more the clean logic directly. What do you think?

Reply:
> Can you point me to code/docs for checking if updates are cleaned up

Yeah I think this is close: https://github.com/solana-labs/solana/blob/v1.7.9/runtime/src/accounts_db.rs#L8138-L8155. You'll have to read out the account index and check the slot list to see the update is gone I think.

> Also, maybe it makes sense to put this as a test in AccountsDb, because these checks don't sound related to snapshots to me, more the clean logic directly. What do you think?

Agreed, they should go in `AccountsDb`

Reply:
Done!